akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -511,9 +530,23 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  private def adjustEventLimit(shell: GraphInterpreterShell): Unit =
+    if (effectiveEventLimit < shell.shellEventLimit) effectiveEventLimit = shell.shellEventLimit
```

Comment:
the idea here is that  in case event limit (that we define as constant in configuration) is less than shell  limit we expanding limit to what's for shell. In this case we avoiding additional Resume messages.

We can have limit as 1 only in case we set limit constant for actor in 1 and event limit for all shells is 1.